Gene Table Cleanup - data file updates script

rmadupuri opened the pull request:
Script to propagate changes in gene symbols and entrez gene ids in data files.

### What does the script do?
**Step1:**

Replaces the outdated entrez ids to new entrez ids in data files. 

**Step2:**

Updates the hugo symbols in data files based on entrez ids.

1. If entrez id is valid (i.e, it is in either the gene or gene-alias tables)
- If entrez id is only in the gene table, and the hugo symbol in data file does not correspond to the symbol in the gene table for the given entrez id, update the symbol in data file to the symbol from gene table based on the entrez id.
- If entrez id only in the gene-alias table, and the hugo symbol in data file does not correspond to alias symbol for the given entrez id, update the symbol to the alias symbol. If multiple aliases exist for the entrez id, do nothing for now. Have to look into other profiles of the study for mapping these cases.
- If entrez id is in both the gene and gene-alias tables, and the hugo symbol in data file does not correspond to main/alias symbols for the given entrez id, then update the hugo symbol in data file to the symbol from gene table.

2. If entrez id in data file is invalid (i.e, it is not observed in the gene or gene-alias tables)
- If the hugo symbol in the data file is outdated, update it to new hugo symbol.
- If the hugo symbol in data file is `NA`, update the hugo symbol in file to empty cell. `NA` in our vocabulary means Unknown, but in the gene tables it maps to entrez id `7504`. So during the import, there is a chance of incorrectly mapping these records to `7504` due to importer falling back on hugo symbols when entrez id in not known.

**Step3:**

Replaces the outdated hugo symbols to new hugo symbols in data files, if the file has only hugo symbol column.

Every change that is made to data files is logged for reference.

yichaoS commented on HGNC_Genomic_Files_Migration/README.md:
```diff
@@ -0,0 +1,31 @@
+## HGNC Genomic Data Migration Tool
+
+This tool is intended to migrate the changes in entrez ids and hugo symbols based on HGNC in data files. The tool,
```
"intended to update data files in accordance to <link_to_seed_DB_update>"

rmadupuri replied:
Updated. Pointed the news release link to Gene Table Updates Readme for now. Will have to change the links once the location of the finalized/PR is merged.

yichaoS commented on HGNC_Genomic_Files_Migration/README.md:
```diff
@@ -0,0 +1,31 @@
+## HGNC Genomic Data Migration Tool
+
+This tool is intended to migrate the changes in entrez ids and hugo symbols based on HGNC in data files. The tool,
+1. Replaces the outdated entrez ids/hugo symbols to new entrez ids/hugo symbols. The outdated to new entrez id/hugo symbols mappings are pre-defined based on our analysis and is included in the script folder.
```
"The ... mappings are actively maintained/updated..".

rmadupuri replied:
Done

yichaoS commented on HGNC_Genomic_Files_Migration/README.md:
````diff
@@ -0,0 +1,31 @@
+**Options**
+```
+-path | --source_path: Path to the data file or directory (of files) that needs to be migrated.
+-l | --stdout_log: Dry-run the script. Preview the changes that will be made to the files.
````
The "-l" could be skipped here, and instead add this description into overall script description.

yichaoS commented on HGNC_Genomic_Files_Migration/hgnc_data_file_migration.py:
```diff
@@ -0,0 +1,243 @@
+#---Check the input type (file/directory) and get the list of files (1 or many)--->
+def check_path(source_path):
+	files_list = []
+	exluded_files_list = ['data_bcr_clinical_data_patient.txt','data_bcr_clinical_data_sample.txt','data_clinical_patient.txt','data_clinical_sample.txt','data_clinical_supp_hypoxia.txt','data_gene_matrix.txt','data_microbiome.txt','data_mutational_signature_confidence.txt','data_mutational_signature_contribution.txt','data_timeline_labtest.txt','data_timeline_procedure.txt','data_timeline_specimen. ...
```
`data_fusion` files should be updated as well for outdated symbols?

rmadupuri replied:
Updated the script to handle `data_fusions`. The symbols in `Fusion` column gets updated as well along with `Hugo_Symbol` column.

yichaoS commented on HGNC_Genomic_Files_Migration/README.md:
```diff
@@ -0,0 +1,31 @@
+2. Updates the non-matching hugo symbols based on the entrez ids. 
+For example, if the data file contains hugo symbol entrez id pair `MST4	6788`, the hugo symbol `MST4` gets updated to `STK3` since MST4 is not a valid symbol for entrez id 6788.
+
+### Prerequisites
```
Change to "Requirements"

rmadupuri replied:
Done

yichaoS commented:
1. Are the 2 outdated lists align with the lists in the spread sheet? https://docs.google.com/spreadsheets/d/1KbgGWRV9DsRmQozA_pWHOjTq3Am5OewXWQR8SyVxUlE/edit#gid=0
2. Are the 2 HGNC tables align with the tables in the spreadsheet above?
3. Remove the word "HGNC" from the tables and scripts names - since the updates we added are not all from HGNC, the majority are from NCBI or current DB. 
4. The descriptions of the PR above describes the logic of the script, which is useful and could be included as a separate section in the readme as well.

yichaoS commented on HGNC_Genomic_Files_Migration/hgnc_data_file_migration.py:
```diff
@@ -0,0 +1,243 @@
+				#If entrez id is in outdated list - update the entrez id to new entrez id.
+				if entrez in outdated_entrez_dict:
+					log += entrez+'\t\t---entrez id replaced to---\t\t'+outdated_entrez_dict[entrez]+'\n'
+					line1[entrez_index] = entrez = outdated_entrez_dict[entrez]
```
`line1` could be updated to a better variable name.

rmadupuri replied:
Done

rmadupuri commented:
Thank you for the review @yichaoS. 
1. Yes, the two outdated lists align to the spread sheet.
2. Yes, the two HGNC tables align with the tables in the spreadsheet.
3. Good point. Removed the use of word `HGNC`.
4. Added the logic to README.
